### PR TITLE
[#4959] feat(Iceberg): support Iceberg REST extend API

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?

support Iceberg REST extend API

### Why are the changes needed?

Fix: #4959 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
1.  add new operation classes with new URI PATH to `org.apache.gravitino.iceberg.service.rest2`
2. config `gravitino.iceberg-rest.extension-packages` to  `org.apache.gravitino.iceberg.service.rest2`
3. start Gravitino IcebergRESTServer
4. check new URI is accessable.

### CODE REVIEW

**Comment by jerryshao** [comment]
I think you'd better discuss with the Pinterest team about the solutions before proposing it, to see if it can satisfy their needs. Moreover, it would be better to let them do it.

**Comment by FANNG1** [comment]
@SinghAsDev this PR add a configuration to extend the REST API packages, is it meet your requirement?

**Comment by FANNG1** [comment]
> Hi @FANNG1 thanks for the change, I think it's in the right direction. Can you help add a test case that adds an extension uri, and run some test against it.

ok

**Comment by puchengy** [comment]
@FANNG1 Thanks for sharing this. I am curious in this PR, how can users access `icebergCatalogWrapperManager` instance if user plugin their API into IRC?

**Comment by FANNG1** [comment]
> @FANNG1 Thanks for sharing this. I am curious in this PR, how can users access `icebergCatalogWrapperManager` instance if user plugin their API into IRC?

1.  `IcebergCatalogWrapperManager` injected to jersey in `RESTService.java`

```java
    icebergCatalogWrapperManager = new IcebergCatalogWrapperManager(icebergConfig.getAllConfig());
    icebergMetricsManager = new IcebergMetricsManager(icebergConfig);
    config.register(
        new AbstractBinder() {
          @Override
          protected void configure() {
            bind(icebergCatalogWrapperManager).to(IcebergCatalogWrapperManager.class).ranked(1);
            bind(icebergMetricsManager).to(IcebergMetricsManager.class).ranked(1);
          }
        });
```

2.  The user plugin could use the injected `IcebergCatalogWrapperManager` like `IcebergTableOperations`
```java
  @Inject
  public IcebergTableOperations(
      IcebergCatalogWrapperManager icebergCatalogWrapperManager,
      IcebergMetricsManager icebergMetricsManager) {
    this.icebergCatalogWrapperManager = icebergCatalogWrapperManager;
    this.icebergObjectMapper = IcebergObjectMapper.getInstance();
    this.icebergMetricsManager = icebergMetricsManager;
  }
```

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/extension/HelloOperations.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.gravitino.iceberg.service.IcebergRestUtils;
+
+@Path("/hello")
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/extension/HelloOperations.java:32]: Couldn't use `HELLO_URI_PATH` here

**Comment by FANNG1** [comment]
Add a test to produce "hello" message, please help to review this when you are free.   @SinghAsDev @puchengy @jerryshao

**Review by jerryshao — APPROVED** [review]

**Pull request merged**